lib/Client.js

```diff
@@ -77,12 +77,34 @@ class Client extends EventEmitter {
                     recipients: body.d.recipients
                 };
 
-                house.rooms = [...house.rooms, {}];
+                house.rooms = [...house.rooms, {room}];
 
                 this.rooms.collect(room.id, room);
                 this.homes.set(house.id, house);
             }
 
+            if (body.e == 'ROOM_UPDATE') {
+                // Room has been edited
+
+                this.emit(body.e, body.d);
+            }
+
+            if (body.e == 'ROOM_DELETE') {
+                // Room deleted, remove it from our cache
+                let rooms = this.houses.get(body.d.house_id).rooms;
+
+                for(let i = 0; i > rooms.length(); i++) {
```

Comment:
To make this simpler, you can do:

`let rooms = this.houses.get(body.d.house_id).rooms.filter(r => r.id !== body.d.id)`

To filter out the deleted room from the rooms array.